NEXT-9067 - Update MailService.php

vttn opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/community/contribution-guideline?category=shopware-platform-dev-en/community).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?
We are trying to stop emails being sent to customers until our payment handler webhook responds with an 'paid' status.

### 2. What does this change do, exactly?
Stop emails being sent out if MailBeforeValidateEvent propagation is stopped.

### 3. Describe each step to reproduce the issue or behaviour.
Listen to the MailBeforeValidateEvent, and stop propagation

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-9067

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

nlubisch commented on tests/Functional/Core/sBasketTest.php:
```diff
@@ -1804,6 +1804,102 @@ public function testsGetBasket()
+        $resourceHelper = new \Shopware\Tests\Functional\Bundle\StoreFrontBundle\Helper();
+        try {
+            // Setup article for the first basket position - an article that costs EUR 29.97
+            $article = $resourceHelper->createArticle(array(
```
use short array syntax

fixpunkt replied:
This would make this test inconsistent with the rest of the test case.

nlubisch commented on tests/Functional/Core/sBasketTest.php:
```diff
@@ -1804,6 +1804,102 @@ public function testsGetBasket()
+     */
+    public function testsGetBasketDataNegativeCloseToZeroTotal()
+    {
+        $resourceHelper = new \Shopware\Tests\Functional\Bundle\StoreFrontBundle\Helper();
```
import class

samuelvogel replied:
There are no imports in this test class right now and Shopware does it the same way [here](https://github.com/shopware/shopware/blob/v5.2.20/tests/Functional/Core/sBasketTest.php#L2300) as well, so it should probably be fine for a single use.

marcop73 commented on engine/Shopware/Core/sBasket.php:
```diff
@@ -1214,6 +1214,26 @@ public function sGetBasketData()
+     * @param float|double $total a number to round
+     * @return double the number rounded to two decimal places, with -0.0 replaced with 0.0
+     */
+    private static function roundTotal($total)
```
IMHO Is a private function instead of a private **static** function more suitable

fixpunkt replied:
Thanks for your feedback.

Since it doesn't need `$this`, it is static by definition. The keyword makes this explicit. What's your reasoning for why this should be an instance method?

htkassner commented:
Thank you for your contribution. This will be released with our 5.3 version.

vttn committed:
Update MailService.php

NEXT-9067 - Stop email being send out if MailBeforeValidateEvent is stop propagation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1027) <br/>All committers have signed the CLA.

vttn committed:
Update MailService.php

Remove whitespace

jkrzefski commented:
I have already issued a [pull request for this problem](https://github.com/shopware/platform/pull/720) with a bit more of a global scope. Instead of only stopping mail sending, the entire business event is cancelled when propagation was stopped. I actually did not take the time to check, whether this fixed your exact problem, but it looks suspiciously similar to me.

vttn commented:
@jkrzefski this is a different issue.

jkrzefski commented:
@vttn Alright.

lernhart commented:
Thank you for your contribution @vttn :tada: :+1:
